core/unix/os.c

```diff
@@ -8525,8 +8548,10 @@ get_dynamo_library_bounds(void)
     ASSERT(check_start == dynamo_dll_start);
     dynamo_dll_end = check_end;
 #else
-    dynamo_dll_start = check_start;
-    dynamo_dll_end = check_end;
+    if (!have_linker_vars) {
```

Comment:
Hmm a little confusing -- the have_linker_vars portion should be just like the first #if and maybe should do that assert?  Should we combine them here (i.e., just #if MACOS #else) and set have_linker_vars for all Linux?

Reply:
I was originally thinking we couldn't do the assert (since the whole point is to workaround the fact that the memquery function doesn't correctly find the bounds for our function), but on second thought it should be correct assuming that the dynamo_so_start is actually pointing to the first ELF header: the header-parsing logic will discover the full image size and things should work correctly.

I've combined the logic and included the ASSERT for when we have the linker vars.